menas/src/main/scala/za/co/absa/enceladus/menas/services/DatasetService.scala

```diff
@@ -154,13 +154,13 @@ class DatasetService @Autowired()(datasetMongoRepository: DatasetMongoRepository
       .foldLeft(Validation.empty){(acc, propDef)=>
         propDef.essentiality match {
           case Mandatory(true) if forRun =>
-            log.warn(s"Property '${propDef.name}' is required to be present, but was not found! (Run enabled)")
-            acc
+            acc.withWarning(propDef.name,
+              s"""Property '${propDef.name}' is required to be present, but was not
+                 | found! This warning will turn into error after the transition period""".stripMargin.replace("\n", ""))
```

Comment:
This message is little Absa specific. I don't mind if we agree to keep the version from develop in 2.23.0

Reply:
Wait, the same wording is there.. Totally missed that. Actually not tested such a property 😊